azure-pipelines/templates/steps/generate-sbom.yml

```diff
@@ -0,0 +1,37 @@
+# File: azure-pipelines\templates\steps\generate-sbom.yml
+# Description: Tasks to generate SBOM manifest file
+# Parameters:
+#  'project' name of the project to generate SBOM
+#  'buildDropPath' path to folder for project's build artifacts
+#  'configuration' gradle configuration to lock dependencies for
+#  'artifactName' artifact name to publish dependencies lockfile
+
+parameters:
+- name: project
+- name: buildDropPath
+- name: configuration
+- name: artifactName
+  default: dependencies
+
+steps:
+- task: Gradle@3
+  displayName: generate dependencies lockfile
+  inputs:
+    tasks: ${{ parameters.project }}:dependencies --configuration=${{ parameters.configuration }} --write-locks
```

Comment:
Does this include transitive dependencies?

Reply:
yes